git-link.el

```diff
@@ -471,6 +482,15 @@ return (FILENAME . REVISION) otherwise nil."
 
       (list host path))))
 
+(defun git-link--ssh-resolve-hostname (hostname)
+  "Resolve HOSTNAME using ssh client."
+  (let ((output (shell-command-to-string (format "ssh -G %s" hostname)))
+	(host nil))
+    (dolist (line (string-split output "\n"))
```

Comment:
`string-split`, where does this come from? I do not have it in my Emacs. Using `27.2`

Reply:
Also on Window will this be `\n` or `\r\n`?

Reply:
It comes from subr.el which comes with Emacs 29.1. I will see what I can do to support older versions.

Reply:
What about this? 

```el
(defun find-it (host)
  (with-temp-buffer
    (call-process "/usr/bin/ssh" nil (list (current-buffer) nil) nil "-G" host)
    (beginning-of-buffer)
    (when (search-forward-regexp "^hostname \\(.*\\)")
      (match-string 1))))
```
Should take care of the Windows line terminator problem. If it was even a problem.

Reply:
Well `(defun git-link--ssh-resolve-hostname ...)` 😉

Reply:
Should also just be `"ssh"`, right?

Reply:
@sshaw Sorry, this took more time to come back than what I expected. I have replaced `string-split` with `split-string` which is available from Emacs 20.1. In fact `string-split` is a `defalias` to it.

Reply:
Hi, great thanks for the update!

Do we know `\n` will work on Windows? It's okay if not. 

Can you squash into a single commit?

Also added one question about `cond`

Reply:
> Do we know \n will work on Windows? It's okay if not.

It won't work. I can make it portable if you want, but I wouldn't be able to test it on a Windows machine since I don't have one.

> Can you squash into a single commit?

Done, thanks!

Reply:
>> Do we know \n will work on Windows? It's okay if not.

> It won't work. I can make it portable if you want, but I wouldn't be able to test it on a Windows machine since I don't have one.

And I just uninstalled my Virtual Box Windows (7!) image. Should just be a matter of changing `"\n"` to `"\r?\n"` but we can do that latter. 

Thanks for the work!